File: 2_Programação_sistemas_operacionais/2.1_File_stdio/Python/Ex8.py
import struct

def abre_arq(arquivo, modo):
	try:
		p = open(arquivo, modo)
	except OSError:
		print("Erro! Impossivel abrir o arquivo!")
		exit(-1)
	return p

pi = 3.1415
fp = abre_arq("arquivo.bin", "w")
fp.write("pi = %f" % pi)
fp.close()
fp = abre_arq("arquivo.bin", "r")
c1,c2,pilido = fp.readline().split()
print("O valor de PI (lido do arquivo) é", pilido)
fp.close()

File: 2_Programação_sistemas_operacionais/2.1_File_stdio/Ex8.c
#include <stdio.h>
#include <stdlib.h>
#include "file_stdio.h"

int main()
{
	FILE *fp;
	float pi = 3.1415;
	float pilido;

	fp = abre_arq("arquivo.txt", "w");
	fprintf(fp, "pi = %f\n", pi);
	fclose(fp);

	fp = abre_arq("arquivo.txt", "r");
	fscanf(fp, "pi = %f\n", &pilido);
	printf("O valor de PI "
		"(lido do arquivo) eh: %f\n",
		pilido);
	fclose(fp);
	return(0);
}
